packages/blockchain/src/index.ts

```diff
@@ -71,14 +71,21 @@ export interface BlockchainOptions {
   db?: LevelUp
 
   /**
-   * This flags indicates if Proof-of-work should be validated. Defaults to `true`.
+   * This flags indicates if a block should be validated by the consensus algorithm
+   * or protocol used by the chain.
+   *
+   * Supported: 'pow' with 'ethash' algorithm (taken from the `Common` instance)
+   * Default: `true`.
    */
-  validatePow?: boolean
+  validateConsensus?: boolean
```

Comment:
I am slightly confused here what the difference is between `validateConsensus` and `validateBlocks`. So if we use PoW and verify Consensus then it checks if the ethash of the block is correct? What does `validateBlocks` then do? Does it also imply that we verify the consensus, or can we validate the block but not the consensus? `validateBlocks` would throw on things like some parameters which are incorrect (like a timestamp which is less than the parent block)?

Reply:
Yes, I have to say that I am somewhat confused by this as well and I have to say that I don't have a good overview which kind of these validation checks are absolutely necessary and which ones are just a nice-to-have thing. I think we'll get some more intuitive understanding on this over time when dealing more with integrating the `blockchain`, `block` and `tx` libraries in the client. Actually love that this work triggers theses kind of questions.

I think one crucial basic part is the underdeveloped documentation here. On this round I will concentrate on this and go through the validation methods and give this some more clarity. Not sure yet, it might even be worth to rename the one or the other method, some good names can sometimes really do miracles on these kind of things.

Reply:
This was actually a super-good question which triggered direct changes in the code, thanks! 😄

Reply:
I think in general, this is the idea: `validatePow` was used to validate the mining work, while `validateBlock` is to essentially validate everything else in the block (the header is correct, the uncles are correct, parent hash exists in the chain, etc.). I assume this is because sometimes you just want to execute a block and skip the PoW check since it is very expensive to come up with a correct PoW. Also in terms of the test suite. `ethereum/tests` also has many tests which have `NoProof`, which implies that you check everything except PoW.